src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -59,19 +68,82 @@ def get_action_type(alert_rule_trigger_action: AlertRuleTriggerAction) -> str |
         return Action.Type.EMAIL
 
 
-def migrate_metric_action(
-    alert_rule_trigger_action: AlertRuleTriggerAction,
-) -> tuple[Action, DataConditionGroupAction] | None:
+def get_detector_trigger(
+    alert_rule_trigger: AlertRuleTrigger, priority: DetectorPriorityLevel
+) -> DataCondition | None:
+    """
+    Helper method to find the detector trigger corresponding to an AlertRuleTrigger.
+    Returns None if the detector cannot be found. Raises an exception if the detector
+    exists but the detector trigger cannot be found.
+    """
+    alert_rule = alert_rule_trigger.alert_rule
+    try:
+        alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=alert_rule)
+    except AlertRuleDetector.DoesNotExist:
+        # We attempted to dual delete a trigger that was not dual migrated
+        return None
+
+    detector = alert_rule_detector.detector
+    detector_data_condition_group = detector.workflow_condition_group
+    if detector_data_condition_group is None:
+        logger.error(
+            "detector_data_condition does not exist",
+            extra={"alert_rule_trigger_id": alert_rule_trigger.id},
+        )
+        raise MissingACITableException
+
     try:
-        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
-            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        detector_trigger = DataCondition.objects.get(
+            condition_group=detector_data_condition_group,
+            condition_result=priority,
         )
-    except AlertRuleTriggerDataCondition.DoesNotExist:
+    except DataCondition.DoesNotExist:
         logger.exception(
-            "AlertRuleTriggerDataCondition does not exist",
-            extra={"alert_rule_trigger_id": alert_rule_trigger_action.alert_rule_trigger.id},
+            "detector trigger does not exist",
+            extra={"alert_rule_trigger_id": alert_rule_trigger.id},
         )
-        return None
+        raise MissingACITableException
+    return detector_trigger
+
+
+def get_action_filter(
+    alert_rule_trigger: AlertRuleTrigger, priority: DetectorPriorityLevel
+) -> DataCondition:
+    """
+    Helper method to find the action filter corresponding to an AlertRuleTrigger.
+    Raises an exception if the action filter cannot be found.
+    """
+    alert_rule = alert_rule_trigger.alert_rule
+    try:
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(alert_rule=alert_rule)
+    except AlertRuleWorkflow.DoesNotExist:
+        logger.exception(
+            "workflow does not exist",
+            extra={"alert_rule_trigger_id": alert_rule_trigger.id},
+        )
+        raise MissingACITableException
```

Comment:
should we just invoke the `AlertRuleWorkflow.objects.get` method rather than wrapping it in try catch then raising a new error?

in general, it's frowned upon to catch than re-raise because you'll lose the stack trace. in this case, it's p clear _why_ this exception would be raised though.